src/Runner.Sdk/ProcessInvoker.cs

```diff
@@ -264,7 +264,19 @@ public ProcessInvoker(ITraceWriter trace)
             {
                 foreach (KeyValuePair<string, string> kvp in environment)
                 {
+#if OS_WINDOWS
+                     if(!String.IsNullOrWhiteSpace(kvp.Key) && !String.IsNullOrWhiteSpace(kvp.Value))
```

Comment:
The value could be null correct? IIRC we can set a key to a null value which would just unset the env. 

We may need to do something like

string tempKey = String.IsNullOrWhitespace(kvp.key) ? kvp.key : kvp.Key.Split("\0")[0];
string tempValue // same as above
then set it in the _proc.StartInfo

Reply:
What happens if the initial character is '\0'?

Reply:
If Key is `\0` we will skip it, if value is `\0` we will store key:empty